plugins/woocommerce-admin/client/core-profiler/index.tsx

```diff
@@ -0,0 +1,337 @@
+
+/**
+ * Internal dependencies
+ */
+import { IntroOptIn } from './pages/IntroOptIn';
+import { UserProfile } from './pages/UserProfile';
+import { BusinessInfo } from './pages/BusinessInfo';
+import { BusinessLocation } from './pages/BusinessLocation';
+
+/** Uncomment below to display xstate inspector during development */
+// import { inspect } from '@xstate/inspect';
+// inspect( {
+// 	// url: 'https://stately.ai/viz?inspect', // (default)
+// 	iframe: false, // open in new window
+// } );
+
+// TODO: Typescript support can be improved, but for now lets write the types ourselves
+// https://stately.ai/blog/introducing-typescript-typegen-for-xstate
+
+export type IntroOptInEvent =
+	| { type: 'INTRO_COMPLETED'; payload: { optInDataSharing: boolean } } // can be true or false depending on whether the user opted in or not
+	| { type: 'INTRO_SKIPPED'; payload: { optInDataSharing: false } }; // always false for now
+
+export type UserProfileEvent =
+	| {
+			type: 'USER_PROFILE_COMPLETED';
+			payload: {
+				userProfile: CoreProfilerStateMachineContext[ 'userProfile' ];
+			}; // TODO: fill in the types for this when developing this page
+	  }
+	| {
+			type: 'USER_PROFILE_SKIPPED';
+			payload: { userProfile: { skipped: true } };
+	  };
+
+export type BusinessInfoEvent = {
+	type: 'BUSINESS_INFO_COMPLETED';
+	payload: {
+		businessInfo: CoreProfilerStateMachineContext[ 'businessInfo' ];
+	};
+};
+
+export type BusinessLocationEvent = {
+	type: 'BUSINESS_LOCATION_COMPLETED';
+	payload: {
+		businessInfo: CoreProfilerStateMachineContext[ 'businessInfo' ];
+	};
+};
+
+export type ExtensionsEvent = {
+	type: 'EXTENSIONS_COMPLETED';
+	payload: {
+		extensionsSelected: CoreProfilerStateMachineContext[ 'extensionsSelected' ];
+	};
+};
+
+export type CoreProfilerStateMachineContext = {
+	optInDataSharing: boolean;
+	userProfile: { foo: { bar: 'qux' }; skipped: false } | { skipped: true };
+	geolocatedLocation: {
+		location: string;
+	};
+	extensionsAvailable: ExtensionList[ 'plugins' ] | [];
+	extensionsSelected: string[]; // extension slugs
+	businessInfo: { foo?: { bar: 'qux' }; location: string };
+};
+
+const Extensions = ( {
+	context,
+	sendEvent,
+}: {
+	context: CoreProfilerStateMachineContext;
+	sendEvent: ( payload: ExtensionsEvent ) => void;
+} ) => {
+	return (
+		// TOOD: we need to fetch the extensions list from the API as part of initializing the profiler
+		<>
+			<div>Extensions</div>
+			<div>{ context.extensionsAvailable }</div>
+			<button
+				onClick={ () =>
+					sendEvent( {
+						type: 'EXTENSIONS_COMPLETED',
+						payload: {
+							extensionsSelected: [ 'woocommerce-payments' ],
+						},
+					} )
+				}
+			>
+				Next
+			</button>
+		</>
+	);
+};
+
+const coreProfilerStateMachineDefinition = createMachine( {
+	id: 'coreProfiler',
+	initial: 'initializing',
+	context: {
+		// these are the default values if the user skips everything
+		optInDataSharing: false,
+		userProfile: { skipped: true },
```

Comment:
I was a bit confused at first to see that the default value of skipped is `true`. Is there any benefit to this?

Reply:
Ditto as above! Just scaffolding for 'safe defaults' instead of 'applied defaults', in the event of programmer error.